async_producer.go

```diff
@@ -652,6 +651,7 @@ func (p *asyncProducer) newBrokerProducer(broker *Broker) *brokerProducer {
 		input:          input,
 		output:         bridge,
 		responses:      responses,
+		stopchan:       make(chan struct{}),
```

Comment:
I wonder if if `context` is better, but reckon that could be way more work.

Reply:
No that's a reasonable suggestion, and I don't think it'd be any more work. Code wise the only difference is we'd select on `<--ctx.Done()` rather than `<--stopchan`

Reply:
up to you, I am fine either way for now.

Reply:
@varun06 ok, please go ahead and merge for now then if you're happy. I might revisit under a separate PR putting context everywhere